src/lib/storage/pos_lists/abstract_pos_list.hpp

```diff
@@ -0,0 +1,89 @@
+#pragma once
+
+#include <memory>
+#include <type_traits>
+#include <utility>
+#include <vector>
+
+#include "types.hpp"
+#include "utils/assert.hpp"
+#include "utils/performance_warning.hpp"
+
+namespace opossum {
+
+class RowIDPosList;
+
+class AbstractPosList : private Noncopyable {
+ public:
+  template <typename PosListType = const AbstractPosList*, typename DereferenceReturnType = RowID>
+  class PosListIterator : public boost::iterator_facade<PosListIterator<PosListType, DereferenceReturnType>, RowID,
+                                                        boost::random_access_traversal_tag, DereferenceReturnType> {
+   public:
+    PosListIterator(PosListType pos_list, ChunkOffset offset) {
+      _pos_list = pos_list;
```

Comment:
That looks very much like a pos list copy or am I wrong?
Why not use initializer list?
const parameters.

Reply:
PosListType was a pointer, so the PosList is not copied. We made that more explicit. Also did the rest.